WordPress/Classes/Services/MediaService.m

```diff
@@ -269,6 +269,50 @@ - (void)updateMedia:(Media *)media
                 failure:failureBlock];
 }
 
+- (void)updateMultipleMedia:(NSArray<Media *> *)mediaObjects
+             overallSuccess:(void (^)())overallSuccess
+                    failure:(void (^)(NSError *error))failure
+{
+    if (mediaObjects.count == 0) {
+        if (overallSuccess) {
+            overallSuccess();
+        }
+        return;
+    }
+
+    NSNumber *totalOperations = @(mediaObjects.count);
+    __block NSUInteger completedOperations = 0;
+    __block NSUInteger failedOperations = 0;
+
+    void (^individualOperationCompletion)(BOOL success) = ^(BOOL success) {
+        @synchronized (totalOperations) {
+            completedOperations += 1;
+            failedOperations += success ? 0 : 1;
+
+            if (completedOperations >= totalOperations.unsignedIntegerValue) {
+                if (overallSuccess && failedOperations != totalOperations.unsignedIntegerValue) {
+                    overallSuccess();
+                } else if (failure && failedOperations == totalOperations.unsignedIntegerValue) {
+                    NSError *error = [NSError errorWithDomain:WordPressComRestApiErrorDomain
```

Comment:
@SergioEstevao I added a condition that if all of the operations fail that the error completion handler is called. This matches my original intended behavior as described in the header of the class.